contrib/cirrus/setup_environment.sh

```diff
@@ -99,11 +99,29 @@ esac
 if ((CONTAINER==0)); then  # Not yet running inside a container
     # Discovered reemergence of BFQ scheduler bug in kernel 5.8.12-200
     # which causes a kernel panic when system is under heavy I/O load.
-    # Previously discovered in F32beta and confirmed fixed. It's been
-    # observed in F31 kernels as well.  Deploy workaround for all VMs
-    # to ensure a more stable I/O scheduler (elevator).
-    echo "mq-deadline" > /sys/block/sda/queue/scheduler
-    warn "I/O scheduler: $(cat /sys/block/sda/queue/scheduler)"
+    # Disable the I/O scheduler (a.k.a. elevator) for all environments,
+    # leaving optimization up to underlying storage infrastructure.
+    testfs="/"  # mountpoint that experiences the most I/O during testing
+    msg "Querying block device owning partition hosting the '$testfs' filesystem"
+    # btrfs filesystems append their internal label to the source name
+    # e.g. /dev/nvmeblah1p1[root] - filter this out
+    testdev=$(findmnt --canonicalize --noheadings --output source \
+              --mountpoint $testfs | cut -d '[' -f 1)
+    msg "    found partition: '$testdev'"
+    testdisk=$(lsblk -noheadings --output pkname --paths $testdev)
+    msg "    found block dev: '$testdisk'"
+    testsched="/sys/block/$(basename $testdisk)/queue/scheduler"
+    if [[ -e "$testsched" ]]; then
+        msg "    Found active I/O scheduler: $(cat $testsched)"
+        if [[ ! "$(<$testsched)" =~ \[none\]  ]]; then
+            msg "    Disabling elevator for '$testsched'"
+            echo "none" > "$testsched"
+        else
+            msg "    Elivator already disabled"
+        fi
+    else
+        warn "Sys node for elevator doesn't exist!"
```

Comment:
Helpful hint: including the path ("sys node '$testsched'") will be much appreciated by someone debugging problems in this. Possibly even you :)

Reply:
Good idea, okay it's in (haven't pushed yet).